### PR TITLE
[33685]Number of watchers not shown when opening work package in full screen (through double-click)

### PR DESCRIPTION
**Pull request by bsatarnejad**
watchers number should be displayed in wp full view.

https://community.openproject.com/projects/openproject/work_packages/33685/activity

### CODE REVIEW

File: frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import {Component, Input, OnInit} from '@angular/core';
 import {WorkPackageCacheService} from '../../work-packages/work-package-cache.service';
 import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
+import { WorkPackageWatchersService } from 'core-app/components/wp-single-view-tabs/watchers-tab/wp-watchers.service';
+import { HalResource } from 'core-app/modules/hal/resources/hal-resource';
```
Comment (openprojectci) [inline frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts:6]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/c5a3b40832edc2e18e17e1adb4f8b2994ca923e8/frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts#L6"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts [hunk 2]
```diff
@@ -19,7 +22,10 @@ export class WorkPackageWatchersCountComponent extends UntilDestroyedMixin imple
       .pipe(
         this.untilDestroyed()
       ).subscribe((workPackage) => {
-      this.count = _.size(workPackage.watchers.elements);
+        this.wpWatcherService.require(workPackage)
+        .then((watchers:HalResource[]) => {
+          this.count = watchers?.length;
```
Comment (HDinger) [inline frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts]: I don't think that the `?` is necessary here.

**Review by HDinger — COMMENTED** [review]
@bsatarnejad functionality looks good 👍 Please add a test case to avoid that the bug occurs in the future again. Please also have a look at the failing tests. At least the `watcher_tab_spec.rb` might be related.

**Review by HDinger — APPROVED** [review]
👍 Waiting for travis

**Comment by HDinger** [comment]
Merging as the failing test is unrelated

**Pull request merged**